src/app/[locale]/components/CreateTask.tsx

```diff
@@ -0,0 +1,209 @@
+"use client";
+import { useForm } from "react-hook-form";
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import { useTranslations } from "next-intl";
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch } from "../../../redux/store";
+import { createTask, selectTask } from "../../../redux/slices/taskSlice";
+import { getUsers, selectUser } from "../../../redux/slices/userSlice";
+import { useRouter } from "next/navigation";
+import { selectAuth } from "../../../redux/slices/authSlice";
+import { selectGroup } from "../../../redux/slices/groupSlice";
+import { UserData } from "../types/typescript";
+
+export default function CreateTask({ locale }: { locale: string }) {
+  const t = useTranslations("translations");
+
+  const dispatch = useDispatch<AppDispatch>();
+  const { createTaskId } = useSelector(selectTask);
+  const { users } = useSelector(selectUser);
+  const { isLoggedIn } = useSelector(selectAuth);
+  const [notify, setNotify] = useState<boolean>(false);
+  const { groups } = useSelector(selectGroup);
+  const [filteredUsers, setFilteredUsers] = useState<UserData[]>([]);
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm({
+    defaultValues: {
+      title: "",
+      description: "",
+      type: "",
+      createdOn: "",
+      status: "",
+      assignedTo: "",
+      group: "",
+    },
+  });
+
+  useEffect(() => {
+    if (isLoggedIn === null) {
```

Comment:
I think that this should be handled on page level or better on the set of pages, for example, all pages inside (authenticated) should be available for logged in users only

Reply:
solved based on page level